sklearn/hmm.py

```diff
@@ -623,65 +626,76 @@ class GaussianHMM(_BaseHMM):
         Number of states.
 
-    ``_covariance_type`` : string
-        String describing the type of covariance parameters to
+    algorithm : string, one of the decoder_algorithms, default "viterbi"
+        Decoder algorithm to be used.
         use.  Must be one of 'spherical', 'tied', 'diag', 'full'.
-        Defaults to 'diag'.
-
-    Attributes
-    ----------
-    ``_covariance_type`` : string
+ 
+    init_params : string, string.ascii_letters (all possible parameters)
+        Controls which parameters are initialized prior to
+        training.  Can contain any combination of 's' for
+        startprob, 't' for transmat, 'm' for means, and 'c' for
+        covars.  Defaults to all parameters.
+   
+    covariance_type : string, default 'diag'
         String describing the type of covariance parameters used by
         the model.  Must be one of 'spherical', 'tied', 'diag', 'full'.
 
-    n_features : int
-        Dimensionality of the Gaussian emissions.
-
-    n_components : int
+    n_components : int, default 1
         Number of states in the model.
 
-    transmat : array, shape (`n_components`, `n_components`)
+    transmat : array, shape (`n_components`, `n_components`), default None
         Matrix of transition probabilities between states.
 
-    startprob : array, shape ('n_components`,)
+    startprob : array, shape ('n_components`,), default None
         Initial state occupation distribution.
 
-    means : array, shape (`n_components`, `n_features`)
-        Mean parameters for each state.
-
-    covars : array
-        Covariance parameters for each state.  The shape depends on
-        ``_covariance_type``::
-
-            (`n_components`,)                   if 'spherical',
-            (`n_features`, `n_features`)              if 'tied',
-            (`n_components`, `n_features`)           if 'diag',
-            (`n_components`, `n_features`, `n_features`)  if 'full'
-
     random_state: RandomState or an int seed (0 by default)
         A random number generator instance
 
-    n_iter : int, optional
+    n_iter : int, default 10
         Number of iterations to perform.
 
-    thresh : float, optional
+    thresh : float, default 1e-1
         Convergence threshold.
 
-    params : string, optional
+    params : string, string.ascii_letters (all possible parameters)
         Controls which parameters are updated in the training
         process.  Can contain any combination of 's' for startprob,
         't' for transmat, 'm' for means, and 'c' for covars.
         Defaults to all parameters.
 
-    init_params : string, optional
-        Controls which parameters are initialized prior to
-        training.  Can contain any combination of 's' for
-        startprob, 't' for transmat, 'm' for means, and 'c' for
-        covars.  Defaults to all parameters.
+    covars_prior : float, default 1e-2
+        FIXME
 
+    means_weight : int, default 0
+        FIXME
+
+    covars_weight : int, default 1
+        FIXME
```

Comment:
I don't know what the ``weight``s do, sorry. These are in the deprecated HMM. I don't think we should delay this PR because of these.